tests/testthat/test-as_gt.R

```diff
@@ -0,0 +1,290 @@
+  gt_tbl <- tbl |> as_gt()
+
+  expect_equal(
+    tbl$table_styling$header |>
+      dplyr::filter(!hide) |>
+      dplyr::pull(column),
+    gt_tbl$`_boxhead` |>
+      dplyr::filter(type != "hidden") |>
+      dplyr::pull(var)
+  )
+})
+
+test_that("as_gt passes table column alignment correctly", {
+  expect_equal(
+    my_tbl_regression$table_styling$header |>
+      dplyr::filter(!hide) |>
+      dplyr::pull(align),
+    gt_tbl_regression$`_boxhead` |>
+      dplyr::filter(type != "hidden") |>
+      dplyr::pull(column_align)
+  )
+
+  # customize alignment
+  tbl <- my_tbl_regression |>
+    modify_table_styling(columns = "estimate", align = "right")
+  gt_tbl <- tbl |> as_gt()
+
+  expect_equal(
+    tbl$table_styling$header |>
+      dplyr::filter(!hide) |>
+      dplyr::pull(align),
+    gt_tbl$`_boxhead` |>
+      dplyr::filter(type != "hidden") |>
+      dplyr::pull(column_align)
+  )
+})
+
+test_that("as_gt passes table text interpreters correctly", {
+  # header
+  expect_equal(
+    lapply(
+      my_spanning_tbl$table_styling$header$interpret_label,
+      \(x) do.call(eval(parse(text = x)), list("")) |> class()
+    ),
+    lapply(gt_spanning_tbl$`_boxhead`$column_label, class)
+  )
+
+  # spanning header
+  expect_equal(
+    sapply(
+      my_spanning_tbl$table_styling$header |>
+        dplyr::filter(!is.na(spanning_header)) |>
+        dplyr::pull(interpret_spanning_header),
+      \(x) do.call(eval(parse(text = x)), list("")) |> class()
+    ),
+    gt_spanning_tbl$`_spanners`$spanner_label[[1]] |> class() |>
+      rep(length(gt_spanning_tbl$`_spanners`$vars[[1]])),
+    ignore_attr = "names"
+  )
+
+  # customize interpreter
+  tbl <- my_spanning_tbl |>
+    modify_table_styling(columns = "stat_1", label = "Stat I", spanning_header = "MyTest", text_interpret = "html")
+  gt_tbl <- tbl |> as_gt()
+  if ("html" %in% names(attributes(gt_tbl$`_spanners`$spanner_label[[1]]))) {
+    attr(gt_tbl$`_spanners`$spanner_label[[1]], "html") <- c("html", "character")
+  }
+
+  # header
+  expect_equal(
+    lapply(
+      tbl$table_styling$header$interpret_label,
+      \(x) do.call(eval(parse(text = x)), list("")) |> class()
+    ),
+    lapply(gt_tbl$`_boxhead`$column_label, class)
+  )
+
+  # spanning header
+  expect_equal(
+    sapply(
+      tbl$table_styling$header |>
+        dplyr::filter(!is.na(spanning_header)) |>
+        dplyr::pull(interpret_spanning_header),
+      \(x) do.call(eval(parse(text = x)), list("")) |> class()
+    ),
+    attributes(gt_tbl$`_spanners`$spanner_label[[1]]),
+    ignore_attr = "names"
+  )
+})
+
+test_that("as_gt passes table footnotes & footnote abbreviations correctly", {
+  # footnote
+  expect_equal(
+    my_tbl_summary$table_styling$footnote$column,
+    gt_tbl_summary$`_footnotes`$colname |> unique()
+  )
+  expect_equal(
+    my_tbl_summary$table_styling$footnote$footnote,
+    gt_tbl_summary$`_footnotes`$footnotes |> unlist() |> unique()
+  )
```

Comment:
Can we also test the scenario where a user replaces the default footnotes?

Reply:
Added!